haystack_experimental/components/extractors/llm_metadata_extractor.py

```diff
@@ -240,28 +245,73 @@ def from_dict(cls, data: Dict[str, Any]) -> "LLMMetadataExtractor":
             deserialize_secrets_inplace(data["init_parameters"]["generator_api_params"], keys=["api_key"])
         return default_from_dict(cls, data)
 
+    def _extract_metadata_and_update_doc(self, document, errors, content):
+        """
+        Extract metadata from the content and update the document with the extracted metadata. If the extraction fails,
+        the error message will be stored in `llm_errors`.
+
+        :param document: Document to be updated with the extracted metadata.
+        :param errors: Dictionary to store error messages if the extraction fails.
+        :param content: Content to extract metadata from.
+        """
+        prompt_with_doc = self.builder.run(
+            template=self.prompt,
+            template_variables={self.prompt_variable:content}
```

Comment:
I'm a little confused by this line, do we need to set `template_variables`?

Reply:
Ahh okay I gotcha we are mapping `content` to `self.prompt_variable`. Apologies, missed that point.

Reply:
no need to apologize, it's good that you are thorough on the reviews 👍🏽